plex/player.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#  Copyright (C) 2019 Sascha Montellese <montellese@kodi.tv>
+#
+#  SPDX-License-Identifier: GPL-2.0-or-later
+#  See LICENSES/README.md for more information.
+#
+import threading
+
+import xbmc
+import xbmcmediaimport
+
+from plex.api import Api
+from plex.constants import PLEX_PROTOCOL, PLEX_PLAYER_PLAYING, PLEX_PLAYER_PAUSED, PLEX_PLAYER_STOPPED
+from plex.server import Server
+
+from lib.utils import log, mediaProvider2str
+
+REPORTING_INTERVAL = 5 # seconds
+
+class Player(xbmc.Player):
+    def __init__(self):
+        '''Initializes the player'''
+        super(xbmc.Player, self).__init__()
+
+        self._providers = {}
+        self._monitor = xbmc.Monitor()
+        self._lock = threading.Lock()
+
+    def AddProvider(self, mediaProvider):
+        '''Adds a media provider to the player'''
+        if not mediaProvider:
+            raise ValueError('invalid mediaProvider')
+
+        self._providers[mediaProvider.getIdentifier()] = mediaProvider
+
+    def RemoveProvider(self, mediaProvider):
+        '''Removes the associated media provider'''
+        if not mediaProvider:
+            raise ValueError('invalid mediaProvider')
+
+        del self._providers[mediaProvider.getIdentifier()]
+
+    def onPlayBackStarted(self):
+        '''Triggered when xbmc.Player is started'''
+        self._reset()
+        self._file = self.getPlayingFile()
+
+    def onAVStarted(self):
+        '''Triggered when playback actually starts'''
+        self._startPlayback()
+        self._syncPlaybackState(time=0, state=PLEX_PLAYER_PLAYING)
```

Comment:
`time=0` is not correct if the video is resumed from a specific position.

Reply:
good catch